projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx

```diff
@@ -39,6 +56,28 @@ export default function VideoPressEdit( { attributes, setAttributes } ): React.R
 		setAttributes( { id, src: url } );
 	}
 
+	const videoPressUrl = getVideoPressUrl( guid, {
+		autoplay,
+		controls,
+		loop,
+		muted,
+		playsinline,
+		preload,
+		seekbarColor,
+		seekbarLoadingColor,
+		seekbarPlayedColor,
+		useAverageColor,
+		poster,
+	} );
+
+	// Get video preview status.
+	const { preview, isRequestingEmbedPreview } = usePreview( videoPressUrl );
+
+	// Pick video properties from preview.
+	const { html: previewHtml, scripts } = preview;
+
+	const html = previewHtml || cacheHtml;
```

Comment:
Regarding `cacheHtml` attribute, we'd need to replicate the same logic we have in the web version ([reference](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L160-L177)).

Reply:
I'm not sure this is entirely true. 

I did see the effect that sets the cacheHtml attribute in the web version, which you referenced,  but I don't see where the attribute is actually being used in the video player component.  For now I would suggest we keep the current logic scoped to what is needed to render the video player.

That said I find the comments above the effect hook puzzling and makes wonder if the `cacheHtml` attribute is actually needed. The comment mentions setting [`html` and `thumbnail` attributes](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L162), neither of which are actually set in the effect hook. 
I'll do some commit spelunking to find the reasoning for the setting the `cacheHtml` attribute, perhaps it is needed for the some of the other components like `<VideoPressUploader>` or `<PlaceholderWrapper>`

Reply:
> I did see the effect that sets the cacheHtml attribute in the web version, which you referenced, but I don't see where the attribute is actually being used in the video player component. For now I would suggest we keep the current logic scoped to what is needed to render the video player.

I think the purpose of the `cacheHtml` attribute is to be used upon the component being mounted to display the video player while the preview is being fetched, as it's mentioned in this comment:
https://github.com/Automattic/jetpack/blob/46d3698a631903c49531e264dcb18f35816d7d40/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L184-L186

I understand that this is an enhancement to avoid displaying a loading indicator, which we might consider implementing in the native version. In any case, I tested this functionality in the web version and it's true that is hard to tell the difference with and without having the cache, especially as the HTML content is an iframe that still needs to load the video player content 🤔.

Reply:
> I think the purpose of the cacheHtml attribute is to be used upon the component being mounted to display the video player while the preview is being fetched

Seems like the `const html  = previewHtml || cacheHtml` assignment covers the mounting period in the block lifecycle. 

This enhancement might be for when the preview is fetched _after_ the component is mounted. The hook logic suggests that the `previewHtml` would have to exist then not exist before the `setAttribute` call would have any relevance.  

I'm guessing that happens when the video source is updated. If so, we should probably handle that when we get to settings or uploading updates.